marshalparser/marshalparser.py

```diff
@@ -213,9 +213,13 @@ def read_object(self) -> Any:
         try:
             self.record_object_result(result)
         except UnboundLocalError:
-            raise RuntimeError(
-                f"Error: type [{type}] is recognized but result is not present"
-            )
+            message = f"type [{type}] is recognized but result is not present."
+            if not self.python_version:
+                message += (
+                    "\nThe error is probably caused by an unknown "
+                    "Python version (magic number) if it's a pyc file."
+                )
+            raise RuntimeError(message)
```

Comment:
I was surprised to see UnboundLocalError. Maybe this can help?

```suggestion
            raise RuntimeError(message) from None
```

Reply:
It can, yes, thank you.